### ISSUE TITLE
Split Pipe

### ISSUE THREAD

**Issue by kumarnadarajah** [neslerel/QGISRed#22]
Hello neslerel,
I am using QGISRed 0.10 with QGIS 3.10.0. This is a great plugin to help build an EPANet program from GIS.

I cannot seem to make the Split / Join Pipes function work. I just get the attached error message the pipe is still not split. Can you please help????

By the way, how do I do the pipe joining using this function please????

Many thanks in advance

Kumar

![1 FunctionSelect](https://user-images.githubusercontent.com/66180543/83336873-447d5200-a2b7-11ea-972b-13bc63007930.png)
![2 PipeSelect](https://user-images.githubusercontent.com/66180543/83336879-45ae7f00-a2b7-11ea-91a8-ba555c9e5551.png)
![3 Error](https://user-images.githubusercontent.com/66180543/83336880-46471580-a2b7-11ea-8ec7-dcbd5f76c23b.png)

**Comment by neslerel** [comment]
Hi @kumarnadarajah,

Thanks for your comments!

About the error, I'm not sure which is the cause. If you don't have problem, you could send me the project (layers in a ZIP file) to debug the code and find the error.
Are you tested the tool in different pipes? Which is your Coordinate Reference System (CRS)?

The tool is very simple. If you click on a pipe, it should create an intermediate junction. However, if you click on a junction, the pipes should be joined. In this last case, there are some limitations: Pipes should be the same diameter, material and installation date. Moreover, if the junction selected has a demand, the pipes can not be joined.
![2020-05-31_11-18-38](https://user-images.githubusercontent.com/14369655/83348983-5e5b7b00-a331-11ea-8abe-385fd569a991.gif)

**Comment by kumarnadarajah** [comment]
Hello neslerel,
Many thanks for your quick reply!! This is more or less the first time I had asked for help in a technical forum and I am very pleasantly surprised with the speed and effectiveness of the reply.

It is so strange but I tried it today and it is working!! I recreated the model again from my original GIS shapefiles. I also did have some plugins that were not working which I removed - I noticed the error messages were pointing to these plugins instead of QGISRed .... I did notice that I still get errors when I tried to do things quickly so it is probably a memory thing.

I must again thank you for a very useful plugin for a water engineer like me. I am going to be using it for training my engineers to modify the GIS of our existing network as part of training in hydraulic modelling so thank you on their behalf too I guess.

Best regards,
Kumar Nadarajah
